src/test/suite/didactUriCompletionItemProviderTest.test.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import { expect } from 'chai';
+import { SnippetString } from 'vscode';
+import { DidactUriCompletionItemProvider, DIDACT_COMMAND_PREFIX } from "../../didactUriCompletionItemProvider";
+import { getContext } from '../../extensionFunctions';
+import * as vscode from 'vscode';
+import * as path from 'path'
+
+suite("Didact URI completion provider tests", function () {
+
+	const ctx = getContext();
+	const provider = new DidactUriCompletionItemProvider(ctx);
+
+	test("that all commands in the didactCompletionCatalog.json are available", async () => {
+		const catalog : any = provider.getCompletionCatalog(ctx);
+		const vsCommands : string[] = await vscode.commands.getCommands(true);
+		for (let index = 0; index < catalog.length; index++) {
+			const completion = catalog[index];
+			const fullCommandId = completion.fullCommandId;
+			console.log(`-- ${fullCommandId} is ${vsCommands.includes(fullCommandId)}`);
+			expect(vsCommands.includes(fullCommandId)).to.be.true;
+		}		
+	});
```

Comment:
using a parameterized test will allow to check all commands in a single run even when there are failures. it will also allow to have a better view of which command is missing. it will avoid to have to print it in the console log